Framework/DataHandling/inc/MantidDataHandling/LoadMask.h

```diff
@@ -139,22 +110,22 @@ class DLLExport LoadMask : public API::Algorithm {
 
   /// Default setup.  If true, not masking, but use the pixel
   bool m_defaultToUse;
+  /// input property contains name of instrument definition file rather than
+  /// instrument name itself
+  bool m_IDF_provided;
 
-  std::vector<int32_t> mask_detid_single;
-  std::vector<int32_t> mask_detid_pair_low;
-  std::vector<int32_t> mask_detid_pair_up;
-
-  std::vector<int32_t> unmask_detid_single;
-  std::vector<int32_t> unmask_detid_pair_low;
-  std::vector<int32_t> unmask_detid_pair_up;
+  // detector id-s to mask
+  std::vector<detid_t> m_maskDetID;
+  // spectrum id-s to unmask
+  std::vector<detid_t> m_unMaskDetID;
 
   // spectra mask provided
-  std::vector<int32_t> m_MaskSpecID;
-  // spectra unmask provided
-  std::vector<int32_t> m_unMaskSpecID;
+  std::vector<specnum_t> m_maskSpecID;
+  // spectra unmask provided NOT IMPLEMENTED
+  // std::vector<specnum_t> m_unMaskSpecID;
```

Comment:
Do not know -- I was asking Nick to remove this code (for another ticket) and he suggested to keep it. May be he misunderstood me, but if needs to be removed I have to ask him again. 
this would be another ticket.